### PR TITLE
[DC-347] Populate cdr_metadata table with cdr generation metadata

### PR DESCRIPTION
**Pull request by ksdkalluri**
Signed-off-by: Krishna Kalluri <krishnasaikalluri@gmail.com>

### CODE REVIEW

File: data_steward/tools/add_cdr_metadata.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+                   destination_dataset_id=target_dataset_id,
+                   destination_table_id=METADATA_TABLE)
+
+
+def add_metadata(dataset_id, project_id, field_values=None):
+    """
+
+    :param dataset_id: 
+    :param project_id: 
+    :param field_values:
```
Comment (nishanthpp93) [inline data_steward/tools/add_cdr_metadata.py]: What is `field_values` here? A docstring would help

File: data_steward/tools/add_cdr_metadata.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    """
+    q = ETL_VERSION_CHECK.format(etl=ETL_VERSION,
+                                 dataset=dataset_id,
+                                 table=METADATA_TABLE)
+    etl_check = bq.query(q, project_id=project_id)[ETL_VERSION].tolist()
+    if not etl_check:
+        q = ADD_ETL_METADATA_QUERY.format(dataset=dataset_id,
+                                          etl_version=ETL_VERSION,
+                                          field_value=field_values[ETL_VERSION])
+        bq_utils.query(q)
```
Comment (nishanthpp93) [inline data_steward/tools/add_cdr_metadata.py]: This could use `utils.bq` as well

File: data_steward/tools/add_cdr_metadata.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+NAME = 'name'
+TYPE = 'type'
+DATE = 'date'
+DESCRIPTION = 'description'
+
+JOB_COMPONENTS = [COPY, CREATE, INSERT]
+
+fields_filename = os.path.join(resources.fields_path, METADATA_TABLE + '.json')
+with open(fields_filename, 'r') as fields_file:
+    fields = json.load(fields_file)
```
Comment (nishanthpp93) [inline data_steward/tools/add_cdr_metadata.py]: We have a function for this already: `resources.fields_for()`

File: data_steward/tools/add_cdr_metadata.py [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+                    UPDATE_STRING_COLUMNS.format(
+                        field=field_name, field_value=field_values[field_name]))
+    if len(statement_list) != 0:
+        update_statement = ' ,'.join(statement_list)
+        print(update_statement)
+        q = UPDATE_QUERY.format(dataset=dataset_id,
+                                table=METADATA_TABLE,
+                                statement=update_statement,
+                                etl_version=ETL_VERSION,
+                                etl_value=field_values[ETL_VERSION])
```
Comment (nishanthpp93) [inline data_steward/tools/add_cdr_metadata.py]: Do you mind adding tests to that this works as intended?

Comment (ksdkalluri) [reply data_steward/tools/add_cdr_metadata.py]: Done added tests

File: data_steward/tools/add_cdr_metadata.py [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+    """
+
+    :param source_dataset_id:
+    :param target_dataset_id:
+    :param fields:
+    :return:
+    """
+    create_metadata_table(target_dataset_id, fields)
+    query = COPY_QUERY.format(datset=source_dataset_id)
+    bq_utils.query(query,
```
[the remark this answers is not in the document]
Comment (ksdkalluri) [reply data_steward/tools/add_cdr_metadata.py]: updated

**Pull request merged**